### ISSUE TITLE
I2C communication failure

### ISSUE THREAD

**Issue by commiepinko** [waydabber/m1ddc#14]
I'm running macOS 12.4 on an M1 Mini with a Dell P3222QE.  I get 

`I2C communication failure: (iokit/?) unknown subsystem error`

in response to every m1ddc command.

This is a bit beyond my pay grade - should I assume the problem is with the display?

**Comment by pgoca** [comment]
I have the same problem on my M1 macbook pro with AOC Q27G2G4. I also run it on macOS 12.4. The error itself is way beyond my understanding, so I'm just leaving a note that it's not only your display that fails :)

**Comment by waydabber** [comment]
Yeah, some displays just won't support DDC. It is either the HDMI port of the Mac (which has issues with DDC) or the display. I have a Lenovo Creator Extreme, it does not support DDC over the USB-C connection (even though it uses DisplayPort alt-mode) but supports it over DisplayPort for example. It is totally display firmware specific.

**Comment by pgoca** [comment]
Thanks for the quick response, I will try with the display port

**Comment by x-magic** [comment]
I have a Dell U4320Q and it works fine via USB-C DP Alt port, but won't work via DP on a Thunderbolt dock. I guess it has something to do with whether the middle layers (docks, USB hub controllers) support DDC/CI or not.

**Comment by waydabber** [comment]
yes, it depends on how the dock handles i2c communication for ddc. Some docks/dongles are just not handling it right or doing it only for EDID data (which is required so the display works at all).

**Comment by emircanerkul** [comment]
> I2C communication failure: (iokit/?) unknown subsystem error

@waydabber When I use https://github.com/waydabber/BetterDisplay#readme It works perfectly but when I use this cli, it doesn't. Says same error in my mac m1. I just need a basic HDMI to DP switch command.

**Comment by waydabber** [comment]
Hi, can you try the latest BetterDisplay version (that supports ddc input switch - does that work)?

**Comment by waydabber** [comment]
Resolution, refresh rate, colors (if you are talking about adjustments) and brightnes (by default for the 0-50% part) does not require DDC. Question is if does DDC brightness (normally 50-100% which controls the hardware backlight), DDC contrast, DDC color control, DDC display input switching etc. work with BD. If not, then the problem is the same as with m1ddc.

**Comment by emircanerkul** [comment]
@waydabber yea, you're right. Now I checked the settings under the DDC video adjustment tab, all doesn't work.

**Comment by waydabber** [comment]
What connection are you using?

**Comment by emircanerkul** [comment]
@waydabber Display Port (LS28AG700N)

**Comment by waydabber** [comment]
Well, that should work. :) It might be that the display has some issues or does not support DDC on all ports. I have no experience with the G7.

**Comment by sidevesh** [comment]
Seeing this error to and in my case this happens with all monitors I have tried and I can verify that the same monitors through the same dongle / dock works on my Linux or Windows laptop.
Is there a way we can get more info into the error atleast, the current error is really vague.

**Comment by sidevesh** [comment]
I am also seeing the error on an Intel MacBook, again, another Windows laptop on the same exact config works normally.

**Comment by sidevesh** [comment]
Maybe unrelated but in my Intel MacBook, the MonitorControl app started working after the recent update

**Comment by duncanfinney** [comment]
Just a note for other people facing this issue: Seems to work great with Caldigit TB3 dock, but my Anker 777 doesn't pass through the i2c sadly.

**Comment by lovish-hustle** [comment]
For me, it was neither the monitor nor the mac. It was the cable I was using. I have two type C to Hdmi cables but, it works with one and throws this error with the other. Any idea why?
